cmd/agent/main.go

```diff
@@ -112,6 +125,10 @@ func (o *GrpcProxyAgentOptions) Flags() *pflag.FlagSet {
 	flags.DurationVar(&o.syncIntervalCap, "sync-interval-cap", o.syncIntervalCap, "The maximum interval for the syncInterval to back off to when unable to connect to the proxy server")
 	flags.StringVar(&o.serviceAccountTokenPath, "service-account-token-path", o.serviceAccountTokenPath, "If non-empty proxy agent uses this token to prove its identity to the proxy server.")
 	flags.StringVar(&o.agentIdentifiers, "agent-identifiers", o.agentIdentifiers, "Identifiers of the agent that will be used by the server when choosing agent. N.B. the list of identifiers must be in URL encoded format. e.g.,host=localhost&host=node1.mydomain.com&cidr=127.0.0.1/16&ipv4=1.2.3.4&ipv4=5.6.7.8&ipv6=:::::")
+	flags.StringVar(&o.kubeconfigPath, "kubeconfig", o.kubeconfigPath, "Absolute path to the kubeconfig file. Only required when --update-pod-status is set.")
+	flags.Float32Var(&o.kubeconfigQPS, "kubeconfig-qps", o.kubeconfigQPS, "Maximum client QPS")
+	flags.IntVar(&o.kubeconfigBurst, "kubeconfig-burst", o.kubeconfigBurst, "Maximum client burst")
+	flags.BoolVar(&o.enablePodCondition, "enable-pod-condition", o.enablePodCondition, "When set, manage a condition to represent connection status in the pod specified by POD_NAME and POD_NAMESPACE")
```

Comment:
We should add validation that if enablePodCondition is true that any required fields, such as podname and namespace are set. We should also check at a minimum that kubeconfigPath is set to an existing path and readable.

Reply:
Good call. Added for pod name/namespace. But the kubeconfig is actually optional since `clientcmd.BuildConfigFromFlags` defaults to the in-cluster config.